internal/frrsockets/frrsockets_test.go

```diff
@@ -44,3 +26,61 @@ func TestExecuteCmd(t *testing.T) {
 		t.Fatalf("executeCmd expected '%s', got '%s'\n", expected, resp)
 	}
 }
+
+// TestExecuteCmdWithLargeOutput tests ExecuteCmd when the command returns
+// a large amount of output exceeding the hard-coded buffer size of 4096.
+func TestExecuteCmdWithLargeOutput(t *testing.T) {
+	socketPath := filepath.Join(os.TempDir(), "bgp_mock.vty")
+
+	command := "show a whole lot of data"
+
+	expected := generateRandomString(5000)
```

Comment:
There's no real need for this to be random each time. It can literally be any character other than 0x00, repeated > 4096 times, e.g. `strings.Repeat("z", 5000)`

Reply:
will do. I was not familiar with strings.Repeat.